### DISCUSSION TITLE
Wazuh agents cannot connect to "forward/sensor" node

### DISCUSSION THREAD

**Discussion by licnoya** [Security-Onion-Solutions/securityonion#2724] [Unsupported Versions] [answered]
I have a distributed setup in my lab (Version 2.3.21 installed via ISO image on two virtual machines), on one side is the ManagerSearch node and on the other is the Sensor node. But when I try to connect the Wazuh agent to the Sensor node all I got is time out on the agent side. Looks to me like that port(1514) is not open. I tried so-allow but all I get is some errors that the firewall file is missing, also the same thing with so-allow-view (check the picture). Do I need to copy that file from the manager or do I configure that on the manager? I tried connecting the Wazuh agent to the Manager node and it's working with no problems.

Please help, thank you.

![2021-01-25 20_19_29-OnionSensor on LICNOYAPC - Virtual Machine Connection](https://user-images.githubusercontent.com/77996319/105770145-198af100-5f5f-11eb-9741-d35cbd62a358.png)

**Comment by weslambert** [comment, marked as answer]
For now, you'll need to add the sensor to the host group for Wazuh, as described here:

https://docs.securityonion.net/en/latest/firewall.html?highlight=firewall#examples

**Reply by licnoya** [reply]
Thank you so much, somehow I overlooked this. This solved the problem.

**Reply by innovate-support** [reply]
@licnoya can you post what exactly you did to 'fix' the problem?  I tried running `sudo so-firewall includehost wazuh_agent X.X.X.X` on the Manager which added the Sensor's IP to the /opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml file, but nothing is resolved.  Agent still can't connect to Sensor.

I even tried running `sudo so-allow` again on the Sensor to allow Agent IP, but nothing changed.

**Comment by facyber** [comment]
Hi @weslambert 

I have exact same issue only with Manager and Heavy node. The steps I did on manager node are:

1. so-firewall includehost wazuh_agent <MGMT_IP_HEAVY_NODE>
2. salt <HOSTNAME>_heavynode state.apply firewall
3. Simple `telnet <MANAGER_IP> 1514` from heavy node gives connection refused.
4. I tried dropping all firewall rules on both manager and heavy node, and connection is still not possible on port 1514 not either from end user Wazuh agent to the node, nor from node to manager.

What is that I am doing wrong? Is it expected that on heavy node there is nothing in /opt/so/saltstack/local folder? Did I understood correctly that Manager node is the one responsible for managing firewall rules on all nodes?

**Reply by cm-ops** [reply]
https://github.com/Security-Onion-Solutions/securityonion/discussions/2455#discussioncomment-364087

Follow these steps.
